app/src/main/java/com/patrickgatewood/weather/ui/WeatherPresenter.java

```diff
@@ -0,0 +1,11 @@
+package com.patrickgatewood.weather.ui;
+
+
+public interface WeatherPresenter {
+
+    void onFetchButtonClick();
+
+    void attachView(WeatherActivity weatherActivity);
```

Comment:
You probably don't want to explicitly require an `Activity` here. this makes your interface very inflexible.